src/utils/testsToStatusUpdate.ts

```diff
@@ -9,33 +9,42 @@ export const statusMap: Partial<Record<TestStatus, API_StatusValue>> = {
   [TestStatus.Broken]: "error",
 };
 
-const statusOrder: API_StatusValue[] = ["unknown", "pending", "success", "warn", "error"];
-function chooseWorseStatus(status: API_StatusValue, oldStatus: API_StatusValue | null) {
-  if (!oldStatus) return status;
-
+const statusOrder: (API_StatusValue | null)[] = [
+  null,
+  "unknown",
+  "pending",
+  "success",
+  "warn",
+  "error",
+];
+function chooseWorseStatus(status: API_StatusValue | null, oldStatus: API_StatusValue | null) {
   return statusOrder[Math.max(statusOrder.indexOf(status), statusOrder.indexOf(oldStatus))];
 }
 
 export function testsToStatusUpdate<T extends StatusTestFieldsFragment>(
   tests: readonly T[]
 ): API_StatusUpdate {
-  const storyIdToStatus: Record<StoryId, API_StatusValue> = {};
+  const storyIdToStatus: Record<StoryId, API_StatusValue | null> = {};
   tests.forEach((test) => {
+    if (!test.story || !test.status) {
+      return;
+    }
     storyIdToStatus[test.story.storyId] = chooseWorseStatus(
-      statusMap[test.status],
+      statusMap[test.status] || null,
       storyIdToStatus[test.story.storyId]
     );
   });
   const update = Object.fromEntries(
     Object.entries(storyIdToStatus).map(([storyId, status]) => [
       storyId,
-      {
+      status && {
         status,
         title: "Visual Tests",
         description: "Chromatic Visual Tests",
       },
     ])
   );
 
+  // @ts-expect-error SB's API type is incorrect here you are allowed to pass null
   return update;
```

Comment:
@ndelangen the type in SB's API is wrong. It says the values have to be `API_StatusObjects`:

https://github.com/storybookjs/storybook/blob/04ced719c25b81af8ab11813acf54f856b8fe3f8/code/lib/types/src/modules/api-stories.ts#L186

But then it does something different if they are `null`:

https://github.com/storybookjs/storybook/blob/04ced719c25b81af8ab11813acf54f856b8fe3f8/code/lib/manager-api/src/modules/stories.ts#L599

TBH I am not sure why TS isn't complaining about that line because if you look at the type of `value` it can never be `null` (according to the incorrect type definition):

<img width="636" alt="image" src="https://github.com/chromaui/addon-visual-tests/assets/132554/c9e8bf76-f046-4321-b533-5908ddb55523">

Maybe because we don't have strict types on yet for `manager-api`? https://github.com/storybookjs/storybook/blob/04ced719c25b81af8ab11813acf54f856b8fe3f8/code/lib/manager-api/tsconfig.json#L5

Reply:
Indeed, not yet:
https://github.com/storybookjs/storybook/blob/702fd3e3ebf17e880fd82682a58b3f11e789f0c6/code/lib/manager-api/tsconfig.json#L5

Reply:
I opened a PR with a fix here:
https://github.com/storybookjs/storybook/pull/24206